bin/src/data/csv.py

```diff
@@ -124,7 +124,7 @@ def add_split(self, config: dict,  force=False) -> None:
         train, validation, test = self.experiment.get_function_split(split_method)(self.data, **config['params'])
 
         # add the split column to the data
-        split_column = np.full(len(self.data), np.nan).astype(int)
+        split_column = np.full(len(self.data), -1).astype(int)
```

Comment:
why -1 instead of nan what does this do? coulòd you add a comment?

Reply:
The code you provided  and the warning message "RuntimeWarning: invalid value encountered in cast" arise because you're trying to convert np.nan (Not a Number) to an integer type.

Here's a breakdown of the code:

np.full(len(self.data), np.nan): This part creates a new NumPy array filled with np.nan (representing missing data) with the same length as the self.data attribute of your class MyClass.
.astype(int): This attempts to convert the elements in the newly created array to integers. However, since np.nan cannot be accurately represented as an integer, it raises the RuntimeWarning.